networkx/algorithms/min_edge_cover.py

```diff
@@ -0,0 +1,74 @@
+"""
+An edge cover of a graph is a set of edges such that every vertex of
+the graph is incident to at least one edge of the set.
+A minimum edge cover is an edge covering of smallest possible size.
+
+https://en.wikipedia.org/wiki/Edge_cover
+"""
+#    Copyright (C) 2015 by
+#    Nishant Nikhil <nishantiam@gmail.com>
+#    All rights reserved.
+#    BSD license.
+
+import networkx as nx
+from networkx.algorithms import bipartite
+
+
+def min_edge_cover(G):
+    """
+    A smallest edge cover can be found in polynomial time by finding
+    a maximum matching and extending it greedily so that all vertices
+    are covered.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        Undirected graph
+        Unweighted graph
+
+    Returns
+    -------
+    min_cover : dictionary
+        The matching is returned as a dictionary, min_weighted_edge_cover,
+        such that min_weighted_edge_cover[v] has a list of nodes so that
+        the edge joining them occurs in minimum edge cover.
+        If no edge cover exists, return an empty dictionary.
+
+    Notes
+    -----
+    Worst Case runtime of maximum_matching for bipartite graph is
+    O(number_of_nodes ** 2.5)
+    Worst Case runtime of maximum_matching for other graphs is
+    O(number_of_nodes ** 3)
+    And the greedy extension works in O(number_of_nodes)
+    So worst case run time of the function is
+    O(number_of_nodes ** 2.5) or O(number_of_nodes ** 3) accordingly.
+    """
+    maximum_matching = {}
+    is_cover_exist = True
+    for node in G.nodes():
+        if not G[node]:
+            is_cover_exist = False
+            break
+    if not is_cover_exist:
+        # min_cover does not exist
+        # so return an empty dictionary
+        return maximum_matching
+    if bipartite.is_bipartite(G):
+        # for better run time
+        maximum_matching = bipartite.maximum_matching(G)
+    else:
+        maximum_matching = nx.max_weight_matching(G, maxcardinality=True)
+    min_cover = {}
+    # min_weighted_edge_cover is superset of maximum_matching
+    # but unlike maximum_matching, here a key can have multiple values
+    for key in maximum_matching.keys():
+        min_cover[key] = []
+        min_cover[key].append(maximum_matching[key])
+    # iterate for uncovered nodes
+    for node in G.nodes():
+        if node not in min_cover.keys():
+            min_cover.setdefault(node, [])
+            min_cover[node].append(G[node].keys()[0])
+            min_cover[G[node].keys()[0]].append(node)
```

Comment:
```diff
diff --git a/networkx/algorithms/min_edge_cover.py b/networkx/algorithms/min_edge_cover.py
index 962c4e6..82a507a 100644
--- a/networkx/algorithms/min_edge_cover.py
+++ b/networkx/algorithms/min_edge_cover.py
@@ -69,6 +69,6 @@ def min_edge_cover(G):
     for node in G.nodes():
         if node not in min_cover.keys():
             min_cover.setdefault(node, [])
-            min_cover[node].append(G[node].keys()[0])
-            min_cover[G[node].keys()[0]].append(node)
+            min_cover[node].append(list(G[node].keys())[0])
+            min_cover[list(G[node].keys())[0]].append(node)
     return min_cover
```
This should fix the error for Python 3. `dict.keys()`, `dict.values()` and `dict.items()` don't return a list in Python3 but iterables.